### ISSUE TITLE
Option for "variable-name" rule to allow leading underscore for property backing fields

### ISSUE THREAD

**Issue by glen-84** [palantir/tslint#1489]
It would be nice to have this:

``` ts
class Example {
    private _var1; // ERROR, not a backing field, shouldn't start with an underscore.
    private _var2; // Allowed, as it's a backing field for the `var2` property.

    public get var2() {
        return this._var2;
    }
}
```

`allow-leading-underscore-backing-fields` maybe.

Edit: It would be considered a backing field if there was either a getter or a setter with the same name (sans the leading underscore).

**Comment by stevedenman** [comment]
This naming convention for backing fields is also suggested [in the typescript handbook ](https://www.typescriptlang.org/docs/handbook/classes.html#accessors);

    private _fullName: string;

    get fullName(): string {
        return this._fullName;
    }

**Comment by JoshuaKGoldberg** [comment]
`variable-name` now allows options such as `"allow-leading-underscore"`. 🎉

**Comment by glen-84** [comment]
@JoshuaKGoldberg That's not new, and allows leading underscores on *any* variable.

**Comment by JoshuaKGoldberg** [comment]
Oh! My apologies, I speed-read the rule and missed the difference. Agreed that an option to allow a leading underscore _only_ for backing properties would be nice.

We should note that it shouldn't be allowed to be enabled at the same time as `"allow-leading-underscore"`.

**Comment by Xenya0815** [comment]
The issue is closed but I can't find a option at the documentation. Is it still planned to offer a option like `allow-leading-underscore-backing-fields`?

**Comment by JoshuaKGoldberg** [comment]
Thanks for the catch @Xenya0815 - this should have been re-opened.

**Comment by adidahiya** [comment]
Closing in favor of https://github.com/typescript-eslint/typescript-eslint/issues/816

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
